snapcraft/meta/snap_yaml.py

```diff
@@ -179,22 +178,22 @@ def get_content_dirs(self, installed_path: Path) -> Set[Path]:
 class Links(_SnapMetadataModel):
     """Metadata links used in snaps."""
 
-    contact: Optional[UniqueStrList]
-    donation: Optional[UniqueStrList]
-    issues: Optional[UniqueStrList]
-    source_code: Optional[UniqueStrList]
-    website: Optional[UniqueStrList]
+    contact: Optional[models.UniqueStrList]
```

Comment:
I believe we can use [craft-application's UniqueStrList](https://github.com/canonical/craft-application/blob/83b5cd37973a3e6581b24f89ea178e5d2e860d53/craft_application/models/constraints.py#L54) here.

Reply:
Nice, I switched to this in a few places.  Can you check it?  I'm not sure if the `cast` calls are the right approach

Reply:
You can do them with casts or just by calling the constructor (the constraint types are subclasses of their actual types). I don't think it makes much difference in code speed.